rhoas-client/src/main/java/com/redhat/service/bridge/rhoas/KafkasMgmtV1ClientImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.redhat.service.bridge.rhoas;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.openshift.cloud.api.kas.models.ServiceAccount;
+import com.openshift.cloud.api.kas.models.ServiceAccountRequest;
+
+import io.quarkus.oidc.client.NamedOidcClient;
+import io.quarkus.oidc.client.OidcClient;
+import io.quarkus.oidc.client.Tokens;
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.core.Vertx;
+
+@ApplicationScoped
+public class KafkasMgmtV1ClientImpl extends AbstractAppServicesClientImpl implements KafkasMgmtV1Client {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkasMgmtV1ClientImpl.class);
+
+    @ConfigProperty(name = "event-bridge.rhoas.mgmt-api.host")
+    String basePath;
+    @ConfigProperty(name = "event-bridge.rhoas.sso.red-hat.refresh-token")
+    String refreshToken;
+
+    @Inject
+    Vertx vertx;
+
+    @Inject
+    @NamedOidcClient("red-hat-sso")
+    OidcClient client;
+
+    Tokens tokens;
+
+    @PostConstruct
+    void onPostConstruct() {
+        init(vertx);
+    }
+
+    @Override
+    public Uni<ServiceAccount> createServiceAccount(ServiceAccountRequest serviceAccountRequest) {
+        return securityValueCall(api -> api.createServiceAccount(serviceAccountRequest));
+    }
+
+    @Override
+    public Uni<Void> deleteServiceAccount(String id) {
+        return securityVoidCall(api -> api.deleteServiceAccountById(id));
+    }
+
+    @Override
+    protected String getAccessToken() {
+        if (tokens == null || tokens.isAccessTokenExpired()) {
+            tokens = client.refreshTokens(refreshToken).await().indefinitely();
```

Comment:
Same here

Reply:
I was checking what @r00ta is doing in #198 , but probably we can move on with this and then refactor the token refresh mechanism later once we finalize it.

Reply:
We also shouldn't `await().indefinitely()`. A sensible timeout should be applied with a clear error message if we cannot obtain a token within a period of time.

@kostola How do you want to address this? In this PR or wait for @r00ta work?

Reply:
@robpblake I added a timeout of 30 seconds similar to the one in [ManagerSyncServiceImpl](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/9e6430b2a0493a209e1a1789eee30dbc8c2a4185/shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java#L38).

As for the [other token instance](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/196#discussion_r778630573), let's refactor this in a dedicated PR once we agree on a final refresh mechanism to overcome Quarkus issues.